packages/format-library/src/link/inline.js

```diff
@@ -56,181 +64,55 @@ const URLPopoverAtLink = ( { isActive, addingLink, value, ...props } ) => {
 
-	onKeyDown( event ) {
-		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
-			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
-			event.stopPropagation();
-		}
-	}
-
-	onChangeInputValue( inputValue ) {
-		this.setState( { inputValue } );
-	}
-
-	setLinkTarget( opensInNewWindow ) {
-		const { activeAttributes: { url = '' }, value, onChange } = this.props;
-
-		this.setState( { opensInNewWindow } );
-
-		// Apply now if URL is not being edited.
-		if ( ! isShowingInput( this.props, this.state ) ) {
-			const selectedText = getTextContent( slice( value ) );
-
-			onChange( applyFormat( value, createLinkFormat( {
-				url,
-				opensInNewWindow,
-				text: selectedText,
-			} ) ) );
-		}
-	}
-
-	editLink( event ) {
-		this.setState( { editLink: true } );
-		event.preventDefault();
-	}
-
-	submitLink( event ) {
-		const { isActive, value, onChange, speak } = this.props;
-		const { inputValue, opensInNewWindow } = this.state;
-		const url = prependHTTP( inputValue );
+	const linkValue = {
+		url: activeAttributes.url,
+		opensInNewTab: activeAttributes.target === '_blank',
+	};
+	const onChangeLink = ( newValue ) => {
+		const newUrl = prependHTTP( newValue.url );
 		const selectedText = getTextContent( slice( value ) );
 		const format = createLinkFormat( {
-			url,
-			opensInNewWindow,
+			url: newUrl,
+			opensInNewWindow: newValue.opensInNewTab,
 			text: selectedText,
 		} );
 
-		event.preventDefault();
-
 		if ( isCollapsed( value ) && ! isActive ) {
-			const toInsert = applyFormat( create( { text: url } ), format, 0, url.length );
+			const toInsert = applyFormat( create( { text: newUrl } ), format, 0, newUrl.length );
 			onChange( insert( value, toInsert ) );
 		} else {
 			onChange( applyFormat( value, format ) );
 		}
 
-		this.resetState();
-
-		if ( ! isValidHref( url ) ) {
+		if ( ! isValidHref( newUrl ) ) {
 			speak( __( 'Warning: the link has been inserted but may have errors. Please test it.' ), 'assertive' );
 		} else if ( isActive ) {
 			speak( __( 'Link edited.' ), 'assertive' );
 		} else {
 			speak( __( 'Link inserted.' ), 'assertive' );
 		}
-	}
-
-	onFocusOutside() {
-		// The autocomplete suggestions list renders in a separate popover (in a portal),
-		// so onFocusOutside fails to detect that a click on a suggestion occurred in the
-		// LinkContainer. Detect clicks on autocomplete suggestions using a ref here, and
-		// return to avoid the popover being closed.
-		const autocompleteElement = this.autocompleteRef.current;
-		if ( autocompleteElement && autocompleteElement.contains( document.activeElement ) ) {
-			return;
-		}
 
-		this.resetState();
-	}
-
-	resetState() {
-		this.props.stopAddingLink();
-		this.setState( { editLink: false } );
-	}
-
-	render() {
-		const { isActive, activeAttributes: { url }, addingLink, value } = this.props;
-
-		if ( ! isActive && ! addingLink ) {
-			return null;
+		if ( addingLink ) {
+			stopAddingLink();
```

Comment:
Calling this here ensures that the focus moves back to the RichText somehow 🤷‍♂which is expected in e2e tests.

My personal expectation was that the focus should stay in the popover and potentially when I hit "Escape" it moves back to RichText. I decided to not pursue this right now in this PR because:

 - There's no explicit API for a Format to say "move back to RichText"
 - The RichText autofocus on format hack is still not removed
 - I didn't want to change the behavior compared to master in this refactoring PR.

Reply:
See https://github.com/WordPress/gutenberg/pull/19536. We can address it later.